module_utils/certificate/providers/certmonger.py

```diff
@@ -0,0 +1,136 @@
+import os
+
+from io import StringIO
+
+# pylint: disable=no-name-in-module
+from ansible.module_utils.six.moves.configparser import ConfigParser
+
+from .base import CertificateRequestBaseProvider
+
+
+class CertificateRequestCertmongerProvider(CertificateRequestBaseProvider):
+    """Certmonger provider for certificate Linux System Role."""
+
+    def __init__(self, *args, **kwargs):
+        super(CertificateRequestCertmongerProvider, self).__init__(*args, **kwargs)
+
+        self.requests_dir = "/var/lib/certmonger/requests"
+        self._request_file_path = self._find_certmonger_request_file()
+        self._certmonger_metadata = self._read_certmonger_request_file()
+
+    def _find_certmonger_request_file(self):
+        """Search for certificate metadata in certmonger data directory.
+
+        Looks for the certificate path, file by file inside certmonger data
+        directory.
+
+        Only open files that are 50k or smaller to avoid getting stuck for
```

Comment:
you can remove this comment (or perhaps you already did in a later commit)

Reply:
Nopz! It was lost here. I'll get it removed.

Reply:
36029751c6fcb22a163511a512af21455395304b